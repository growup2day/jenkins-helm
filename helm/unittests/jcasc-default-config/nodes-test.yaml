templates:
  - config-jcasc.yaml
tests:
  - it: should include permanent nodes in default config script when configured
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          commonAttributes:
            mode: NORMAL
            numExecutors: 1
            nodeProperties:
              - toolLocation:
                  locations:
                    - home: "/usr/lib/jvm/java-1.7.0-openjdk-1.7.0.75.x86_64"
                      key: "hudson.model.JDK$DescriptorImpl@JDK 1.7"
                    - home: "/usr/lib/jvm/java-1.8.0"
                      key: "hudson.model.JDK$DescriptorImpl@JDK 1.8"
            remoteFS: "/app/data/jenkins-slave"
            retentionStrategy: "always"
          agents:
            nzdwXXXv:
              host: "nzdwXXXv.nz.unix.test"
              credentialsId: "jenkins-slave-cred"
              labelString: "nzdwXXXv vm maven chrome"
              hostKey: "ssh-rsa host-public-key"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should use remoteFS for the agent if explicitly defined
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          commonAttributes:
            mode: NORMAL
            numExecutors: 1
            nodeProperties:
              - toolLocation:
                  locations:
                    - home: "/usr/lib/jvm/java-1.7.0-openjdk-1.7.0.75.x86_64"
                      key: "hudson.model.JDK$DescriptorImpl@JDK 1.7"
                    - home: "/usr/lib/jvm/java-1.8.0"
                      key: "hudson.model.JDK$DescriptorImpl@JDK 1.8"
            retentionStrategy: "always"
          agents:
            nzdwXX1v:
              host: "nzdwXX1v.nz.unix.test"
              credentialsId: "jenkins-slave-cred"
              labelString: "nzdwXX1v vm maven chrome"
              hostKey: "ssh-rsa host-public-key"
              remoteFS: "/app/data/jenkins-slave"
            nzdwXX2v:
              host: "nzdwXX2v.nz.unix.test"
              credentialsId: "jenkins-slave-cred"
              labelString: "nzdwXX2v vm maven chrome"
              hostKey: "ssh-rsa host-public-key"
              remoteFS: "/app/data/jenkins-slave-other"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should default node labelString to empty when not defined
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          agents:
            nzdwXXXv:
              host: "nzdwXXXv.nz.unix.test"
              credentialsId: "jenkins-slave-cred"
              hostKey: "ssh-rsa host-public-key"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should include the custom node description when nodeDescription is defined
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          agents:
            nzdwXXXv:
              launcherType: "jnlp"
              jnlpTunnel: "openshift-jenkins-jnlp.caas.nz.service.test:30980"
              labelString: "Windows"
              nodeDescription: "Windows slave to run journey test"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should include jnlp configuration when launcherType is jnlp
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          agents:
            nzdwXXXv:
              launcherType: "jnlp"
              jnlpTunnel: "openshift-jenkins-jnlp.caas.nz.service.test:30980"
              labelString: "Windows"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should default ssh agent host and credentialId to empty when they are not defined
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          agents:
            nzdwXXXv:
              labelString: "nzdwXXXv vm maven chrome"
              hostKey: "ssh-rsa host-public-key"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should default jnlp agent tunnel to empty when jnlpTunnel is not defined
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          agents:
            nzdwXXXv:
              labelString: "Windows"
              launcherType: "jnlp"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should include blank nodes config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-nodes.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
  - it: should not include nodes config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            jenkins:
              nodes:
                - permanent:
                    name: nzdwXXXv
    asserts:
      - isNull:
          path: data.[jcasc-default-config-nodes.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
  - it: should include the java path when javaPath is defined
    set:
      JCasC:
        defaultConfig: true
        permanentNodes:
          enabled: true
          agents:
            nzdwXXXv:
              host: "nzdwXXXv.nz.unix.test"
              credentialsId: "jenkins-slave-cred"
              labelString: "nzdwXXXv slave"
              hostKey: "ssh-rsa host-public-key"
              javaPath: "/usr/bin/java"
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-nodes.yaml]
