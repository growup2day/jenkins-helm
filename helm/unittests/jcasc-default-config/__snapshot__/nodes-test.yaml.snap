should default jnlp agent tunnel to empty when jnlpTunnel is not defined:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXXXv"
              nodeDescription: "Permanent node "
              labelString: "Windows"
              launcher:
                jnlp:
                  tunnel: ""
should default node labelString to empty when not defined:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXXXv"
              nodeDescription: "Permanent node nzdwXXXv.nz.unix.test"
              labelString: ""
              launcher:
                ssh:
                  credentialsId: "jenkins-slave-cred"
                  host: "nzdwXXXv.nz.unix.test"
                  port: 22
                  sshHostKeyVerificationStrategy:
                    manuallyProvidedKeyVerificationStrategy:
                      key: "ssh-rsa host-public-key"
should default ssh agent host and credentialId to empty when they are not defined:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXXXv"
              nodeDescription: "Permanent node "
              labelString: "nzdwXXXv vm maven chrome"
              launcher:
                ssh:
                  credentialsId: ""
                  host: ""
                  port: 22
                  sshHostKeyVerificationStrategy:
                    manuallyProvidedKeyVerificationStrategy:
                      key: "ssh-rsa host-public-key"
should include blank nodes config in default config scripts when not configured:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes: []
should include jnlp configuration when launcherType is jnlp:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXXXv"
              nodeDescription: "Permanent node "
              labelString: "Windows"
              launcher:
                jnlp:
                  tunnel: "openshift-jenkins-jnlp.caas.nz.service.test:30980"
should include permanent nodes in default config script when configured:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXXXv"
              nodeDescription: "Permanent node nzdwXXXv.nz.unix.test"
              labelString: "nzdwXXXv vm maven chrome"
              launcher:
                ssh:
                  credentialsId: "jenkins-slave-cred"
                  host: "nzdwXXXv.nz.unix.test"
                  port: 22
                  sshHostKeyVerificationStrategy:
                    manuallyProvidedKeyVerificationStrategy:
                      key: "ssh-rsa host-public-key"
              mode: NORMAL
              nodeProperties:
              - toolLocation:
                  locations:
                  - home: /usr/lib/jvm/java-1.7.0-openjdk-1.7.0.75.x86_64
                    key: hudson.model.JDK$DescriptorImpl@JDK 1.7
                  - home: /usr/lib/jvm/java-1.8.0
                    key: hudson.model.JDK$DescriptorImpl@JDK 1.8
              numExecutors: 1
              remoteFS: /app/data/jenkins-slave
              retentionStrategy: always
should include the custom node description when nodeDescription is defined:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXXXv"
              nodeDescription: "Windows slave to run journey test"
              labelString: "Windows"
              launcher:
                jnlp:
                  tunnel: "openshift-jenkins-jnlp.caas.nz.service.test:30980"
should include the java path when javaPath is defined:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXXXv"
              nodeDescription: "Permanent node nzdwXXXv.nz.unix.test"
              labelString: "nzdwXXXv slave"
              launcher:
                ssh:
                  credentialsId: "jenkins-slave-cred"
                  host: "nzdwXXXv.nz.unix.test"
                  port: 22
                  javaPath: "/usr/bin/java"
                  sshHostKeyVerificationStrategy:
                    manuallyProvidedKeyVerificationStrategy:
                      key: "ssh-rsa host-public-key"
should use remoteFS for the agent if explicitly defined:
  1: |
    |-
      jenkins:
        # Manage Nodes and Clouds > Manage Nodes
        nodes:
          - permanent:
              name: "nzdwXX1v"
              nodeDescription: "Permanent node nzdwXX1v.nz.unix.test"
              labelString: "nzdwXX1v vm maven chrome"
              launcher:
                ssh:
                  credentialsId: "jenkins-slave-cred"
                  host: "nzdwXX1v.nz.unix.test"
                  port: 22
                  sshHostKeyVerificationStrategy:
                    manuallyProvidedKeyVerificationStrategy:
                      key: "ssh-rsa host-public-key"
              remoteFS: "/app/data/jenkins-slave"
              mode: NORMAL
              nodeProperties:
              - toolLocation:
                  locations:
                  - home: /usr/lib/jvm/java-1.7.0-openjdk-1.7.0.75.x86_64
                    key: hudson.model.JDK$DescriptorImpl@JDK 1.7
                  - home: /usr/lib/jvm/java-1.8.0
                    key: hudson.model.JDK$DescriptorImpl@JDK 1.8
              numExecutors: 1
              retentionStrategy: always
          - permanent:
              name: "nzdwXX2v"
              nodeDescription: "Permanent node nzdwXX2v.nz.unix.test"
              labelString: "nzdwXX2v vm maven chrome"
              launcher:
                ssh:
                  credentialsId: "jenkins-slave-cred"
                  host: "nzdwXX2v.nz.unix.test"
                  port: 22
                  sshHostKeyVerificationStrategy:
                    manuallyProvidedKeyVerificationStrategy:
                      key: "ssh-rsa host-public-key"
              remoteFS: "/app/data/jenkins-slave-other"
              mode: NORMAL
              nodeProperties:
              - toolLocation:
                  locations:
                  - home: /usr/lib/jvm/java-1.7.0-openjdk-1.7.0.75.x86_64
                    key: hudson.model.JDK$DescriptorImpl@JDK 1.7
                  - home: /usr/lib/jvm/java-1.8.0
                    key: hudson.model.JDK$DescriptorImpl@JDK 1.8
              numExecutors: 1
              retentionStrategy: always
